TramsDataApi/Program.cs

```diff
@@ -1,25 +1,39 @@
-using Microsoft.AspNetCore.Hosting;
-using Microsoft.Extensions.Hosting;
+using Microsoft.ApplicationInsights.Extensibility;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Logging;
+using Serilog;
+using System;
+using TramsDataApi;
+using TramsDataApi.SerilogCustomEnrichers;
 
-namespace TramsDataApi
+var builder = WebApplication.CreateBuilder(args);
+
+
+var startup = new Startup(builder.Configuration);
+
+startup.ConfigureServices(builder.Services);
+
+builder.Host.UseSerilog((context, services, loggerConfiguration) =>
 {
-    public class Program
-    {
-        public static void Main(string[] args)
-        {
-            CreateHostBuilder(args).Build().Run();
-        }
-
-        public static IHostBuilder CreateHostBuilder(string[] args)
-        {
-            var builder = Host.CreateDefaultBuilder(args);
-            return builder.ConfigureLogging(c =>
-                {
-                    c.ClearProviders();
-                    c.AddConsole();
-                })
-                .ConfigureWebHostDefaults(webBuilder => { webBuilder.UseStartup<Startup>(); });
-        }
-    }
-}
\ No newline at end of file
+    var enricher = services.GetRequiredService<ApiUserEnricher>();
```

Comment:
With the GetRequiredService call it'll throw an exception if the service isn't registered. Is that what we want or do we want some handling of it? I'd guess it probably does constitute a exceptional state, so am happy as is